haskell-backend/src/main/java/org/kframework/backend/haskell/HaskellRewriter.java

```diff
@@ -118,6 +118,10 @@ public RewriterResult execute(K k, Optional<Integer> depth) {
                         args.add("--depth");
                         args.add(options.depth.toString());
                     }
+                    if (options.experimental.smt.smtPrelude != null) {
```

Comment:
I'm fairly sure this will always evaluate to null if you are running kprove and not krun. You want to inject SMTOptions in the constructor above and then access it directly from the injected instance.

Reply:
then again I guess it's fine since this is the execute method...

Reply:
I don't know much about this repo, but this at least works in the case I'm testing.

Reply:
We will definitely need it for kprove, too. @u-- did you try it with `kprove`

Reply:
Anyway, I'm going to approve it so you can test it now, but I might modify it later as we need support for kprove.